From 5e406c848f647bdb0c37fe5ffd553429c7575e51 Mon Sep 17 00:00:00 2001
From: jmbills <jason.m.bills@intel.com>
Date: Fri, 30 Jun 2023 04:23:48 -0600
Subject: [PATCH] Use variable for service install location (#36)

After enabling 'usrmerge', the install location for systemd service
files changed. This uses a variable to get the installation location
instead of hardcoding it.

Tested:
Confirmed that with 'usrmerge' enabled, the service file is correctly
installed.

Signed-off-by: Jason M. Bills <jason.m.bills@intel.com>
Upstream-Status: Pending
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cc6434d..639ed2f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,7 +65,10 @@ else()
         TARGETS cups-service
         DESTINATION sbin)
 
+    find_package (PkgConfig REQUIRED)
+    pkg_get_variable(SYSTEMD_SYSTEM_UNIT_DIR systemd systemdsystemunitdir)
+
     install(
         FILES ${PROJECT_SOURCE_DIR}/xyz.openbmc_project.CupsService.service
-        DESTINATION /lib/systemd/system/)
+        DESTINATION "${SYSTEMD_SYSTEM_UNIT_DIR}")
 endif (UT_ONLY)
-- 
2.25.1

